test/Kubernetes/KubernetesApiExtensionsTests.cs

```diff
@@ -2,11 +2,65 @@
 using k8s.Models;
 using Snd.Sdk.Kubernetes;
 using Xunit;
+using System;
+using System.Linq;
 
 namespace Snd.Sdk.Tests.Kubernetes;
 
 public class KubernetesApiExtensionsTests
 {
+    [Theory]
+    [InlineData("Restart", "Restart", new int[] { 1, 2 }, new int[] { 2, 3 }, 3)]
+    [InlineData("Restart", "Fail", null, null, null)] // Expecting an exception
+    public void MergePodFailurePolicyRules_Test(string actionA, string actionB, int[] exitCodesA, int[] exitCodesB, int? expectedExitCodesCount)
+    {
+        var ruleA = new V1PodFailurePolicyRule(actionA,
+            new V1PodFailurePolicyOnExitCodesRequirement("In", exitCodesA?.ToList()));
+
+        var ruleB = new V1PodFailurePolicyRule(actionB,
+            new V1PodFailurePolicyOnExitCodesRequirement("In", exitCodesB?.ToList()));
+
+        if (expectedExitCodesCount.HasValue)
+        {
+            var result = KubernetesApiExtensions.MergePodFailurePolicyRules(ruleA, ruleB);
+
+            Assert.Equal(actionA, result.Action);
+            Assert.Equal(expectedExitCodesCount.Value, result.OnExitCodes.Values.Count);
+        }
+        else
+        {
+            Assert.Throws<InvalidOperationException>(() => KubernetesApiExtensions.MergePodFailurePolicyRules(ruleA, ruleB));
+        }
+    }
+
+    [Theory]
+    [InlineData("Restart", new int[] { 1, 2, 3 })]
+    public void ConvertToFailurePolicyRule_ValidConversion(string action, int[] exitCodes)
+    {
+        var actionPair = new KeyValuePair<string, List<int>>(action, new List<int>(exitCodes));
+
+        var result = KubernetesApiExtensions.ConvertToFailurePolicyRule(actionPair);
+
+        Assert.Equal(action, result.Action);
+        Assert.Equal(exitCodes.Length, result.OnExitCodes.Values.Count);
+        for (int i = 0; i < exitCodes.Length; i++)
+        {
+            Assert.Contains(exitCodes[i], result.OnExitCodes.Values);
+        }
```

Comment:
collection assert check please :)

Reply:
I'm not sure if it is a "collection" assertion but I've implemented a collection scope assertion in [d21279a](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/d21279af3c03081c2504d45213df2471774025cd)
Do let me know if it was something different that you originally meant